codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -224,9 +233,28 @@ class ServerServiceGeneratorV2(
             "NullabilityChecks" to nullabilityChecks,
             "RoutesArrayElements" to routesArrayElements,
             "SmithyHttpServer" to smithyHttpServer,
+            "PatternInitializations" to patternInitializations(),
         )
     }
 
+    /**
+     * Renders `PatternString::compile_regex()` function calls for every
+     * `@pattern`-constrained string shape in the service closure.
+     */
+    @Suppress("DEPRECATION")
+    private fun patternInitializations(): Writable =
+        Walker(model).walkShapes(service).mapNotNull { shape ->
+            if (shape is StringShape && shape.hasTrait<PatternTrait>() && !shape.hasTrait<software.amazon.smithy.model.traits.EnumTrait>()) {
```

Comment:
nit: import `EnumTrait`.

Reply:
The reason I didn't is to be able to use the `@Suppress("DEPRECATION")` annotation in this function, if you import `EnumTrait` you have to suppress deprecations for the whole file :( (unless there's other syntax I'm missing).

Reply:
You could possibly already use `EnumShape` as `EnumTrait` is deprecated in #1932